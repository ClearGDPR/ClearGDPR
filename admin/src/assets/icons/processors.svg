<svg xmlns="http://www.w3.org/2000/svg" width="13" height="13" viewBox="0 0 13 13">
  <path fill="#1E2028" fill-rule="evenodd" d="M4.64285714,1.39260013 C4.64285714,1.13631899 4.43485714,0.929251263 4.17857143,0.929251263 L1.39285714,0.929251263 C1.13633929,0.929251263 0.928571429,1.13631899 0.928571429,1.39260013 L0.928571429,4.17896109 C0.928571429,4.43477795 1.13633929,4.64347065 1.39285714,4.64347065 L4.17857143,4.64347065 C4.43485714,4.64347065 4.64285714,4.43477795 4.64285714,4.17896109 L4.64285714,1.39260013 Z M4.64285714,5.57156122 L0.928571429,5.57156122 C0.415767857,5.57156122 0,5.15580079 0,4.64347065 L0,0.929251263 C0,0.415992572 0.415767857,0 0.928571429,0 L4.64285714,0 C5.15589286,0 5.57142857,0.415992572 5.57142857,0.929251263 L5.57142857,4.64347065 C5.57142857,5.15580079 5.15589286,5.57156122 4.64285714,5.57156122 L4.64285714,5.57156122 Z M4.64285714,8.82127105 C4.64285714,8.56498991 4.43485714,8.35699363 4.17857143,8.35699363 L1.39285714,8.35699363 C1.13633929,8.35699363 0.928571429,8.56498991 0.928571429,8.82127105 L0.928571429,11.6073999 C0.928571429,11.863681 1.13633929,12.071213 1.39285714,12.071213 L4.17857143,12.071213 C4.43485714,12.071213 4.64285714,11.863681 4.64285714,11.6073999 L4.64285714,8.82127105 Z M4.64285714,13 L0.928571429,13 C0.415767857,13 0,12.5842396 0,12.071213 L0,8.35699363 C0,7.84443135 0.415767857,7.42843878 0.928571429,7.42843878 L4.64285714,7.42843878 C5.15589286,7.42843878 5.57142857,7.84443135 5.57142857,8.35699363 L5.57142857,12.071213 C5.57142857,12.5842396 5.15589286,13 4.64285714,13 L4.64285714,13 Z M12.0714286,1.39260013 C12.0714286,1.13631899 11.8634286,0.929251263 11.6071429,0.929251263 L8.82142857,0.929251263 C8.56514286,0.929251263 8.35714286,1.13631899 8.35714286,1.39260013 L8.35714286,4.17896109 C8.35714286,4.43477795 8.56514286,4.64347065 8.82142857,4.64347065 L11.6071429,4.64347065 C11.8634286,4.64347065 12.0714286,4.43477795 12.0714286,4.17896109 L12.0714286,1.39260013 Z M12.0714286,5.57156122 L8.35714286,5.57156122 C7.84433929,5.57156122 7.42857143,5.15580079 7.42857143,4.64347065 L7.42857143,0.929251263 C7.42857143,0.415992572 7.84433929,0 8.35714286,0 L12.0714286,0 C12.5844643,0 13,0.415992572 13,0.929251263 L13,4.64347065 C13,5.15580079 12.5844643,5.57156122 12.0714286,5.57156122 L12.0714286,5.57156122 Z M12.0714286,8.82127105 C12.0714286,8.56498991 11.8634286,8.35699363 11.6071429,8.35699363 L8.82142857,8.35699363 C8.56514286,8.35699363 8.35714286,8.56498991 8.35714286,8.82127105 L8.35714286,11.6073999 C8.35714286,11.863681 8.56514286,12.071213 8.82142857,12.071213 L11.6071429,12.071213 C11.8634286,12.071213 12.0714286,11.863681 12.0714286,11.6073999 L12.0714286,8.82127105 Z M12.0714286,13 L8.35714286,13 C7.84433929,13 7.42857143,12.5842396 7.42857143,12.071213 L7.42857143,8.35699363 C7.42857143,7.84443135 7.84433929,7.42843878 8.35714286,7.42843878 L12.0714286,7.42843878 C12.5844643,7.42843878 13,7.84443135 13,8.35699363 L13,12.071213 C13,12.5842396 12.5844643,13 12.0714286,13 L12.0714286,13 Z"/>
</svg>
