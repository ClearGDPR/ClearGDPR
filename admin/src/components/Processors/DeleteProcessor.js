import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-responsive-modal';

import Loader from 'components/core/cards/dashboard/Loader';

const DeleteProcessor = props => (
  <Modal open={props.isOpen} showCloseIcon={false} onClose={() => {}} center>
    {!props.isLoading ? (
      <React.Fragment>
        <p>Are you sure you want to delete this processor?</p>
        <p>
          <button
            className="ui-action btn"
            onClick={e => {
              e.preventDefault();
              props.onCancel();
            }}
          >
            Cancel
          </button>
          <button
            className="ui-action btn"
            onClick={e => {
              e.preventDefault();
              props.onConfirm();
            }}
          >
            OK
          </button>
        </p>
      </React.Fragment>
    ) : (
      <Loader />
    )}
  </Modal>
);

DeleteProcessor.propTypes = {
  onConfirm: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  isOpen: PropTypes.bool,
  isLoading: PropTypes.bool
};

export default DeleteProcessor;
