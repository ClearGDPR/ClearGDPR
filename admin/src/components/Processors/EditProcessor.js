import React from 'react';
import PropTypes from 'prop-types';
import { Form, TextArea, Checkbox } from 'react-form';

import TextInput from '../core/cards/dashboard/TextInput';
import Loader from '../core/cards/dashboard/Loader';

// TODO: this should be fetch from configuration service
const DEMO_SCOPES = {
  'Full Name': 'user.fullName',
  Email: 'user.email',
  'Phone Number': 'user.phoneNumber'
};

export class EditProcessor extends React.Component {
  static propTypes = {
    isLoading: PropTypes.bool,
    onSubmit: PropTypes.func,
    touched: PropTypes.object,
    errors: PropTypes.object
  };

  renderForm() {
    return (
      <React.Fragment>
        <TextInput
          label="Name"
          placeholder="*********"
          type={'text'}
          error={
            this.props.touched &&
            this.props.touched['name'] &&
            this.props.errors &&
            this.props.errors['name']
          }
          field="name"
          required
        />

        <label htmlFor="description">Description:</label>
        <TextArea field="description" id="description" />

        <TextInput
          label="Logo URL"
          placeholder="*********"
          type={'text'}
          error={
            this.props.touched &&
            this.props.touched['logoUrl'] &&
            this.props.errors &&
            this.props.errors['logoUrl']
          }
          field="logoUrl"
          required
        />

        <label>Scopes:</label>
        {Object.keys(DEMO_SCOPES).map((scope, i) => <Checkbox key={i} field={`scope[${scope}]`} />)}

        <button type="submit" className="btn">
          Submit
        </button>
      </React.Fragment>
    );
  }

  render() {
    return (
      <form onSubmit={this.props.onSubmit}>
        {this.props.isLoading ? <Loader /> : this.renderForm()}
      </form>
    );
  }
}

const EditProcessorForm = props => (
  <Form onSubmit={submittedValues => props.onSubmit && props.onSubmit(submittedValues)}>
    {formApi => (
      <EditProcessor
        {...props}
        onSubmit={formApi.submitForm}
        errors={{ ...formApi.errors, ...props.errors }}
        touched={formApi.touched}
      />
    )}
  </Form>
);

EditProcessorForm.propTypes = {
  onSubmit: PropTypes.func,
  errors: PropTypes.object
};

export default EditProcessorForm;
