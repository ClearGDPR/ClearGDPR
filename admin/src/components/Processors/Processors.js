import React from 'react';
import PropTypes from 'prop-types';

import { processorType } from 'types';
import ProcessorCard from 'components/core/cards/dashboard/ProcessorCard';
import Loader from 'components/core/cards/dashboard/Loader';
import * as lodash from 'lodash';

const Processors = ({
  processors,
  onCreateProcessorClick,
  onEditProcessorClick,
  onDeleteProcessorClick,
  isLoading,
  children
}) => {
  function onCreateProcessorHandler() {
    onCreateProcessorClick();
  }

  function onEditProcessorHandler(processor) {
    onEditProcessorClick(processor);
  }

  function onDeleteProcessorHandler(processor) {
    onDeleteProcessorClick(processor);
  }

  let renderProcessors = function() {
    return lodash
      .chunk(processors, 3)
      .map(row => (
        <div className="row">
          {row.map((processor, i) => (
            <ProcessorCard
              key={i}
              data={processor}
              onClick={() => onEditProcessorHandler(processor)}
              onDelete={() => onDeleteProcessorHandler(processor.id)}
            />
          ))}
        </div>
      ));
  };

  return (
    <React.Fragment>
      <section className="cards">
        <div className="action-bar">
          <div className="text">
            <h4>Processors</h4>
            <p>
              A processor is a third-party with access to your users data. You can manage from here
              which are enabled or not, and which actions are allowed to access.
            </p>
          </div>
          <div className="spacer" />
          <button className="ui-action btn" onClick={e => onCreateProcessorHandler(e)}>
            + Add Processor
          </button>
        </div>
        {isLoading ? (
          <div className="content">
            <Loader />
          </div>
        ) : (
          renderProcessors()
        )}
      </section>
      {children}
    </React.Fragment>
  );
};

Processors.propTypes = {
  processors: PropTypes.arrayOf(processorType),
  onEditProcessorClick: PropTypes.func.isRequired,
  onCreateProcessorClick: PropTypes.func.isRequired,
  onDeleteProcessorClick: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  children: PropTypes.node
};

export default Processors;
