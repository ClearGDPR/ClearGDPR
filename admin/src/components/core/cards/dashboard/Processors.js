import React from 'react';
import PropTypes from 'prop-types';

import ProcessorCard from './ProcessorCard';

const Processors = props => {
  return (
    <React.Fragment>
      <section className="cards">
        <div className="action-bar row">
          <div className="text">
            <h4>{props.title}</h4>
            <p>{props.desc}</p>
          </div>
          <div className="spacer" />
          <button className="ui-action btn" onClick={props.onClick}>
            + Add Processor
          </button>
        </div>
        <div className="row">
          {props.processors.map((processor, id) => {
            return <ProcessorCard key={id} data={processor} onClick={props.onClick} />;
          })}
        </div>
      </section>
    </React.Fragment>
  );
};

Processors.propTypes = {
  title: PropTypes.string,
  desc: PropTypes.string,
  processors: PropTypes.arrayOf(
    PropTypes.shape({
      logoUrl: PropTypes.string,
      name: PropTypes.string,
      description: PropTypes.string,
      scopes: PropTypes.arrayOf(PropTypes.string)
    })
  ),
  onClick: PropTypes.func
};

export default Processors;
