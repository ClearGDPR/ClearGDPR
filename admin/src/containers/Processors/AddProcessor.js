import React from 'react';
import PropTypes from 'prop-types';

import config from 'config';
import internalFetch from '../../helpers/internal-fetch';
import { PanelConsumer } from '../MainLayout/PanelContext';
import EditProcessorForm from '../../components/Processors/EditProcessor';

export class AddProcessorContainer extends React.Component {
  static propTypes = {
    processor: PropTypes.object,
    closePanel: PropTypes.func
  };

  state = {
    isLoading: false,
    errors: {}
  };

  async updateProcessor(processor) {
    await internalFetch(`${config.API_URL}/api/management/processors/add`, {
      method: 'POST',
      body: JSON.stringify(processor)
    }).catch(err => {
      this.setState({
        errors: {
          '': err.message
        }
      });
      return Promise.reject(err);
    });
  }

  startLoading() {
    this.setState({
      isLoading: true
    });
  }

  stopLoading() {
    this.setState({
      isLoading: false
    });
  }

  onSubmit() {
    this.startLoading();

    return this.updateProcessor(this.props.processor)
      .then(this.stopLoading.bind(this))
      .then(() => this.props.closePanel && this.props.closePanel())
      .catch(this.stopLoading.bind(this));
  }

  render() {
    return (
      <EditProcessorForm
        errors={this.state.errors}
        onSubmit={this.onSubmit.bind(this)}
        isLoading={this.state.isLoading}
      />
    );
  }
}

export default props => (
  <PanelConsumer>
    {({ closePanel }) => <AddProcessorContainer {...props} closePanel={closePanel} />}
  </PanelConsumer>
);
