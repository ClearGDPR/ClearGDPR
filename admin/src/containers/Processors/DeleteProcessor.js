import React from 'react';
import PropTypes from 'prop-types';

import { ProcessorsConsumer } from './ProcessorsContext';
import DeleteProcessor from 'components/Processor/DeleteProcessor';

export class DeleteProcessorContainer extends React.Component {
  static propTypes = {
    processor: PropTypes.object.isRequired,
    deleteProcessor: PropTypes.func.isRequired,
    onClose: PropTypes.func,
    isOpen: PropTypes.bool,
    isLoading: PropTypes.bool
  };

  deleteProcessor() {
    this.props.deleteProcessor(this.props.processor.id).then(this.props.onClose);
  }

  render() {
    return (
      <DeleteProcessor
        onConfirm={this.deleteProcessor.bind(this)}
        onCancel={this.props.onClose}
        isOpen={this.props.isOpen}
        isLoading={this.props.isLoading}
      />
    );
  }
}

export default props => (
  <ProcessorsConsumer>
    {({ deleteProcessor, isLoading }) => (
      <DeleteProcessorContainer {...props} isLoading={isLoading} deleteProcessor={deleteProcessor} />
    )}
  </ProcessorsConsumer>
);
