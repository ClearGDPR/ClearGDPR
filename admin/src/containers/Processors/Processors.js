import React from 'react';
import PropTypes from 'prop-types';

import Processors from 'components/Processors/Processors';
import config from 'config';
import { PanelConsumer } from 'containers/MainLayout/PanelContext';
import Form from 'components/core/cards/dashboard/Form';
import internalFetch from 'helpers/internal-fetch';

export class ProcessorsContainer extends React.Component {
  static propTypes = {
    openPanel: PropTypes.func
  };

  state = {
    processors: [],
    isLoading: true
  };

  async getProcessors() {
    return internalFetch(`${config.API_URL}/api/management/processors/list`);
  }

  componentDidMount() {
    this.getProcessors()
      .then(processors =>
        this.setState({
          processors,
          isLoading: false
        })
      )
      .catch(e => {
        console.error(e);
        this.setState({
          isLoading: false
        });
      });
  }

  onEditProcessorForm() {
    this.props.openPanel(Form);
  }

  render() {
    return (
      <Processors
        processors={this.state.processors}
        isLoading={this.state.isLoading}
        onEditProcessorClick={this.onEditProcessorForm.bind(this)}
      />
    );
  }
}

export default props => (
  <PanelConsumer>
    {({ openPanel }) => <ProcessorsContainer {...props} openPanel={openPanel} />}
  </PanelConsumer>
);
