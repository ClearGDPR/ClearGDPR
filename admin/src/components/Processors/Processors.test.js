import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Processors from './Processors';
import * as ProcessorsDataFactory from 'tests/data/processors.factory';

const setup = propOverrides => {
  const props = Object.assign({
    processors: ProcessorsDataFactory.getAll(),
    isLoading: false,
    onEditProcessorClick: jest.fn(),
    onCreateProcessorClick: jest.fn(),
    onDeleteProcessorClick: jest.fn(),
  }, propOverrides);

  const component = shallow(<Processors {...props} />);

  return { props, component };
};

describe('(Component) Processors', () => {
  it('should render when not loading correctly', () => {
    const { component } = setup();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render when loading correctly', () => {
    const { component } = setup({ isLoading: true });
    expect(toJson(component)).toMatchSnapshot();
  });
});
