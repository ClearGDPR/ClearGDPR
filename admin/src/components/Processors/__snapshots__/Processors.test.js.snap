// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Processors should render when loading correctly 1`] = `
<React.Fragment>
  <section
    className="cards"
  >
    <ActionBar
      desc={
        <React.Fragment>
          A processor is a third-party with access to your users data.
          <br />
          You can manage from here which are enabled or not, and which actions are allowed to access.
        </React.Fragment>
      }
      title="Processors"
    >
      <Component
        onClick={[Function]}
        text="+ Add Processor"
      />
    </ActionBar>
    <div
      className="content"
    >
      <Loader />
    </div>
  </section>
</React.Fragment>
`;

exports[`(Component) Processors should render when not loading correctly 1`] = `
<React.Fragment>
  <section
    className="cards"
  >
    <ActionBar
      desc={
        <React.Fragment>
          A processor is a third-party with access to your users data.
          <br />
          You can manage from here which are enabled or not, and which actions are allowed to access.
        </React.Fragment>
      }
      title="Processors"
    >
      <Component
        onClick={[Function]}
        text="+ Add Processor"
      />
    </ActionBar>
    <div
      className="row"
      key="0"
    >
      <ProcessorCard
        data={
          Object {
            "consented": 72,
            "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum placeat neque, pariatur ipsa qui facere nobis error iste consectetur quo exercitationem!",
            "logoUrl": "nielsen.svg",
            "name": "Nielsen",
            "scopes": Array [
              "Full Name",
              "Email",
              "Phone Number",
            ],
          }
        }
        key="0"
        onClick={[Function]}
        onDelete={[Function]}
      />
      <ProcessorCard
        data={
          Object {
            "consented": 64,
            "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum placeat neque, pariatur ipsa qui facere nobis error iste consectetur quo exercitationem!",
            "logoUrl": "liveramp.svg",
            "name": "Live Ramp",
            "scopes": Array [
              "Email",
            ],
          }
        }
        key="1"
        onClick={[Function]}
        onDelete={[Function]}
      />
      <ProcessorCard
        data={
          Object {
            "consented": 25,
            "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum placeat neque, pariatur ipsa qui facere nobis error iste consectetur quo exercitationem!",
            "logoUrl": "experian.svg",
            "name": "Experian",
            "scopes": Array [
              "Full Name",
              "Phone Number",
            ],
          }
        }
        key="2"
        onClick={[Function]}
        onDelete={[Function]}
      />
    </div>
  </section>
</React.Fragment>
`;
